sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageLockContainer.java

```diff
@@ -3,14 +3,41 @@
 
 package com.azure.messaging.servicebus.implementation;
 
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
 import java.time.Instant;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
- * Container of Message lock and related metadata related.
+ * Container for message locks that are cleaned up periodically.
  */
-public class MessageLockContainer {
-    private final ConcurrentHashMap<String, Instant> lockTokenExpirationMap = new ConcurrentHashMap<>();
+public class MessageLockContainer implements AutoCloseable {
+    private final ClientLogger logger = new ClientLogger(MessageLockContainer.class);
+    private final ConcurrentHashMap<String, Instant> lockTokens = new ConcurrentHashMap<>();
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final Disposable cleanupOperation;
+
+    public MessageLockContainer(Duration cleanupInterval) {
+        cleanupOperation = Flux.interval(cleanupInterval).subscribe(e -> {
+            if (lockTokens.isEmpty()) {
+                return;
+            }
+
+            final Instant now = Instant.now();
+            final boolean removed = lockTokens.entrySet().removeIf(entry -> {
+                final Instant expiration = entry.getValue();
+                return expiration != null && expiration.isBefore(now);
+            });
+
+            if (removed) {
+                logger.verbose("{}: Removed expired entries.", e);
```

Comment:
Maybe worth making this log `info` as removing expired entries should be uncommon and users should be able to see it at info level? Also, consider adding a count of expired entries removed in the log.

Reply:
makes sense. I updated it!